### ISSUE TITLE
Opt-out (or opt-in) warnings when missing an event

### ISSUE THREAD

**Issue by CatThingy** [bevyengine/bevy#7089]
## What problem does this solve or what need does it fill?

Not handling an event can be a source of hard-to-diagnose issues, when the expectation is that all events are handled. This may occur, for example, when events are read with a system governed by a FixedTimestep run criterion. 

Note that all events are necessarily required to be handled. Many events emitted by the engine are of that nature: see #6595. However, events sent from user code usually must be handled to have correct behaviour.

## What solution would you like?

Adding the warning back, but with the ability to opt-out (or opt-in) to the warning for missed events.

## What alternative(s) have you considered?

Always using `ManualEventReader` and checking for missed events.

## Additional context

Previously introduced in #5730 for all events, removed in #6730 as it raised too many false positives.

**Comment by alice-i-cecile** [comment]
Agreed that we want this in some form. It needs to be overrideable though, and can't come at a performance cost in release mode.
